File: weeks/01/6_sparse_arrays/sparse_arrays.rb
def matchingStrings(strings, queries)
    string_counts = {}

    strings.each do |string|
        if string_counts[string]
            string_counts[string] += 1
        else
            string_counts[string] = 1
        end
    end

    query_counts = []

    queries.each do |query|
        if string_counts[query]
            query_counts << string_counts[query]
        else
            query_counts << 0
        end
    end

    query_counts
end

File: weeks/01/6_sparse_arrays/sparse_arrays.py
def matchingStrings(strings, queries):
    
    string_counts = {}
    for string in strings:
        if string not in string_counts:
            string_counts[ string ] = 1
        else:
            string_counts[ string ] += 1
    
    query_counts = []
    for query in queries:
        if query in string_counts:
            query_counts.append( string_counts[query] )    
        else:
            query_counts.append( 0 )

    return query_counts
    